httpx/config.py

```diff
@@ -325,10 +326,25 @@ def __init__(
         if mode not in ("DEFAULT", "CONNECT_ONLY", "TUNNEL_ONLY"):
             raise ValueError(f"Unknown proxy mode {mode!r}")
 
+        if url.username or url.password:
+            headers.setdefault(
+                "Proxy-Authorization",
+                self.build_auth_header(url.username, url.password),
+            )
+            # Remove userinfo from the URL authority, e.g.:
+            # 'username:password@proxy_host:proxy_port' -> 'proxy_host:proxy_port'
+            credentials, _, authority = url.authority.rpartition("@")
+            url = url.copy_with(authority=authority)
```

Comment:
Perhaps replace these two lines with `ul = url.copy_with(username=None, password=None)`?

Reply:
That's definitely cleaner, good catch :+1:

The `copy_with` seems to add the default port to the URL though which forced me to change the tests a bit, but I think that's fine.